### DISCUSSION TITLE
How read InifiniteEnumFlags on Client side (Angular, TS)

### DISCUSSION THREAD

**Discussion by macik1423** [alirezanet/InfiniteEnumFlags#12] [Q&A]
Hi, I've implemented flexible authorization according to [@Json Taylor] (https://github.com/jasontaylordev) presented in [this video](https://www.youtube.com/watch?v=BVJVhceN3N4&list=PLxCsNpHrt57uwt229Frf1tYt2gywF__xg&index=1&t=1265s).
I have only 32 flags because it is enum limitations for TS. My approach is that I have enum permissions in my backend and send permissions as int in JWT token to client. From the client side I have also the same enum permissions and parse this permissions (int) to TS enum. So I can check is user can do something in my application. If I want to implement InfiniteEnumFlags in my app how can I read storing flag id? Am I right that I should send string in JWT permission property (from ToUniqueId method) and read this value in my client but how can client know what this flag means (UniqueId)?

**Comment by alirezanet** [comment]
Hi @macik1423,
Currently, you can't easily parse the UniqueId in the JS world, but a simple solution for this could be adding an endpoint to receive the user permissions from the backend. so the user will log in to your application you will generate a uniqueId and put it in your JWT token then you can use the new endpoint when you need to get the permission names from the backend. 

this endpoint should accept a uniqueId and response with permission names.

**Reply by macik1423** [reply]
Thanks for the quick reply! I will have this solution in my head :), for now 32 flags are enough for my app, but in the future I may need more flags.

**Reply by alirezanet** [reply]
if 32 flags are enough I would also suggest using the built-in enum features but if you think later you might need more than 32 items in your enum it is better to start with InfinateEnumFlags from the beginning because migration from enum to this is not easily possible. (to see the differences I suggest to take a look in the submodule demo I provided)
also, you're welcome, happy to help 💐
